core/src/main/scala/spire/math/Rational.scala

```diff
@@ -690,7 +690,8 @@ object Rational extends RationalInstances {
 
     override def equals(that: Any): Boolean = that match {
       case that: LongRational => this.n == that.n && this.d == that.d
-      case that => super.equals(that)
+      case rat: Rational => rat ==that
+      case that => this == that
```

Comment:
This change looks a bit fishy; possibly prone to endless recursion.

Reply:
Yes I had to revert this. I'm keeping as it was and may add a @nowarn once 2.12 is removed